src/claimtrie.cpp

```diff
@@ -2114,8 +2108,16 @@ bool CClaimTrieCache::incrementBlock(insertUndoType& insertUndo, claimQueueRowTy
             }
             assert(found);
             const std::string name = normalizeClaimName(itEntry->first, forceNormalization);
+
+            // error triggered by claimtriecache_normalization:
+            // it will cause a double remove when rolling back -- something we don't handle correctly
+            // (same problem exists for the insertSupportUndo below)
+            //for (insertUndoType::reverse_iterator itInsertUndo = insertUndo.rbegin(); itInsertUndo != insertUndo.rend(); ++itInsertUndo)
+            //    if (itInsertUndo->name == itEntry->first && itInsertUndo->outPoint == itEntry->second.outPoint)
+            //        throw std::runtime_error("not expected");
```

Comment:
@bvbfan , this situation is for you to look at.

Reply:
I think I understand how this is happening now. I have some ideas on how to address it.

Reply:
Is this still happen? When I correct unit test it does not throw.

Reply:
It fails for me in latest with this test: claimtriebranching_tests/claimtriecache_normalization

Reply:
Did you try that patch:
```
diff --git a/src/test/claimtriebranching_tests.cpp b/src/test/claimtriebranching_tests.cpp
index 48c5ce4f..6072947a 100644
--- a/src/test/claimtriebranching_tests.cpp
+++ b/src/test/claimtriebranching_tests.cpp
@@ -1212,7 +1212,6 @@ BOOST_AUTO_TEST_CASE(claimtriecache_normalization)
     BOOST_CHECK(name != name_upper);
 
     // Add another set of unicode claims that will collapse after the fork
-    CMutableTransaction tx1 = fixture.MakeClaim(fixture.GetCoinbase(), name, "amilie", 2);
     CMutableTransaction tx2 = fixture.MakeClaim(fixture.GetCoinbase(), name_upper, "amelie", 2);
     fixture.MakeClaim(fixture.GetCoinbase(), "amelie1", "amelie", 2);
     fixture.IncrementBlocks(1);
@@ -1231,6 +1230,7 @@ BOOST_AUTO_TEST_CASE(claimtriecache_normalization)
     fixture.setNormalizationForkHeight(1);
     int currentHeight = chainActive.Height();
 
+    CMutableTransaction tx1 = fixture.MakeClaim(fixture.GetCoinbase(), name, "amilie", 2);
     fixture.IncrementBlocks(1);
     // Ok normalization fix the name problem
     BOOST_CHECK(pclaimTrie->getInfoForName(name_normd, nval1));
@@ -1252,9 +1252,8 @@ BOOST_AUTO_TEST_CASE(claimtriecache_normalization)
     BOOST_CHECK(trieCache.spendClaim(name_upper, COutPoint(tx2.GetHash(), 0), currentHeight, amelieValidHeight));
 
     BOOST_CHECK(!pclaimTrie->getInfoForName(name, nval1));
-    BOOST_CHECK(trieCache.getInfoForName(name, nval1));
+    BOOST_CHECK(!trieCache.getInfoForName(name, nval1));
     BOOST_CHECK(trieCache.addClaim(name, COutPoint(tx1.GetHash(), 0), ClaimIdHash(tx1.GetHash(), 0), CAmount(2), currentHeight + 1));
-    BOOST_CHECK(trieCache.getInfoForName(name, nval1));
     insertUndoType insertUndo;
     claimQueueRowType expireUndo;
     insertUndoType insertSupportUndo;
```